lib/hotel.rb

```diff
@@ -0,0 +1,258 @@
+# hotel.rb
+
+# class Room -- do we need a room class or just a reservation one?
+require 'date'
+require 'awesome_print'
+
+# https://ruby-doc.org/stdlib-2.4.1/libdoc/date/rdoc/Date.html
+
+# a single reservations
+class Reservation
+  # attribute accessor
+  attr_accessor :start_date, :end_date, :room_number, :total_cost, :reservation_id, :discount, :block_id
+
+  # initialize
+  def initialize(start_year, start_month, start_day, end_year, end_month, end_day, room_number, cost, discount, block_id)
+    @start_date = Date.new(start_year, start_month, start_day)
+    @end_date = Date.new(end_year, end_month, end_day)
+    @room_number = room_number
+    @cost = cost
+    @total_cost = @cost*(1-discount)*(@end_date. - @start_date).to_i
+    @reservation_id = @start_date.to_time.to_i / (60 * 60 * 24) + room_number #ID is start date as integer plus room number
+    @discount = discount
+    @block_id = block_id
+  end
+end
+
+# the hotel system
+class HotelSystem
+  # attribute accessor
+  attr_accessor :room_numbers, :blocked_rooms, :date_reservation_hash, :id_reservation_hash, :blocks_hash, :date_block_hash
```

Comment:
I see you have several different data structures keeping track of reservations and blocks.  Could this duplication of data prove problematic if you had to make changes later?